src/main/scala/com/microsoft/partnercatalyst/fortis/spark/streamfactories/EventHubStreamFactory.scala

```diff
@@ -5,15 +5,20 @@ import org.apache.spark.streaming.StreamingContext
 import org.apache.spark.streaming.dstream.DStream
 import org.apache.spark.streaming.eventhubs.EventHubsUtils
 
+import java.nio.charset.StandardCharsets
+import scala.language.implicitConversions
 import scala.reflect.ClassTag
+import scala.util.{Failure, Success, Try}
 
-class EventHubStreamFactory[A: ClassTag](
-    progressDir: String,
-    adapter: Array[Byte] => A)
+class EventHubStreamFactory[A: ClassTag](identifier: String, adapter: (Array[Byte]) => Try[A], progressDir: String)
```

Comment:
I think it's important that it's configurable since it's expected to be an HDFS share in production.